src/subdomains/supporting/payment/entities/transaction.entity.ts

```diff
@@ -172,4 +172,8 @@ export class Transaction extends IEntity {
   get refundTargetEntity(): BuyCrypto | BuyFiat | BankTx | undefined {
     return this.buyCrypto ?? this.buyFiat ?? (!this.type && this.bankTx) ?? undefined;
   }
+
+  get completionDate(): Date | undefined {
+    return this.buyCrypto?.outputDate ?? this.buyFiat?.outputDate ?? this.updated;
```

Comment:
Hm but with that, every transaction has a `completionDate`, or? Regardless if the transaction is really completed or not 🤔 For me that doesn't make sense, maybe it's just the naming, but not every transaction is automatically completed I would say